bot/cogs/antimalware.py

```diff
@@ -17,24 +17,29 @@ def __init__(self, bot: Bot):
     @Cog.listener()
     async def on_message(self, message: Message) -> None:
         """Identify messages with prohibited attachments."""
-        rejected_attachments = list()
-        detected_pyfile = list()
+        rejected_attachments = False
+        detected_pyfile = False
         for attachment in message.attachments:
             if not attachment.filename.lower().endswith(tuple(AntiMalwareConfig.whitelist)):
-                rejected_attachments.append(attachment)
+                rejected_attachments = True
             if attachment.filename.lower().endswith('.py'):
```

Comment:
So as I mentioned, this check should come first to make things a bit more efficient. You'd also have to rework the if statements to be like:

`` py
if detected_pyfile:
    ...
elif rejected_attachments:
    ...
```

Reply:
I'm not sure how to implement it quite the way you've shown without either violating DRY or adding another conditional:
```python
if detected_pyfile or rejected_attachments:
    # Send warning message and delete offending message, etc.
```
but I have implemented the re-ordering as you've requested to skip the one extra conditional evaluation in the cases where a .py file is attached. I hope I understood you correctly.

Reply:
That isn't what I meant. Sorry for not being clear. Pseudocode: 

```py
for msg in attachments:
    if py_file:
        detected_pyfile = True
        break
    if not whitelisted:
        rejected_attachments = True
        # Don't break! Since the py message takes priority,
        # we should keep checking attachments to see if maybe
        # a later attachment will be a py file.

if detected_pyfile:
    print("Please use paste service..")
elif rejected_attachments:
    print("File ext not allowed")
else:
    # Do nothing; all attachments sent were whitelisted
    # this else can be omitted but I have shown it here for explanatory purposes
    pass
```

Reply:
Okay, good point, I'll get the second `break` out of there.

Reply:
The conditions for sending out the message still needs to be addressed.

Reply:
Never mind actually, it's fine too. It's just a different way of doing what I had in mind and I didn't recognise that initially.